specification/cognitiveservices/ContentSafety/models.tsp

```diff
@@ -180,3 +258,346 @@ model RemoveTextBlocklistItemsOptions {
+  emphases?: string[];
+  exampleBlobUrl?: url;
+}
+
+#suppress "@azure-tools/typespec-azure-core/documentation-required" "MUST fix in next update"
+@added(ContentSafety.Versions.v2023_10_30_Preview)
+@doc("ImageWithText analyze category.")
+enum ImageWithTextCategory {
+  Hate,
+  SelfHarm,
+  Sexual,
+  Violence,
+}
+
+@added(ContentSafety.Versions.v2023_10_30_Preview)
+@doc("The analysis request of the image with text.")
+model AnalyzeImageWithTextOptions {
+  @doc("The image needs to be analyzed.")
+  image: ImageData;
+
+  @doc("The text attached to the image. We support at most 1k characters (unicode code points) in one text request.")
+  @maxLength(1000)
+  text?: string;
+
+  @doc("The categories will be analyzed. If they are not assigned, a default set of analysis results for the categories will be returned.")
+  categories?: ImageWithTextCategory[];
+
+  @doc("When set to true, our service will perform OCR and concatenate the recognized text with input text before analyzing. We will recognize at most 256 characters (unicode code points) from input image. The others will be truncated.")
+  enableOcr: boolean;
+}
+
+@added(Versions.v2023_10_30_Preview)
+@doc("ImageWithText analysis result.")
+model ImageWithTextCategoriesAnalysis {
+  @doc("The imageWithtext analysis category.")
+  category: ImageWithTextCategory;
+
+  @doc("The higher the severity of input content, the larger this value is. The values could be: 0,2,4,6.")
+  severity?: int32;
+}
+
+@added(Versions.v2023_10_30_Preview)
+@doc("The analysis response of the image with text.")
+model AnalyzeImageWithTextResult {
+  @doc("Analysis result for categories.")
+  categoriesAnalysis: ImageWithTextCategoriesAnalysis[];
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("The text to detect ungroundedness.")
+model DetectUngroundednessInputContent {
+  @doc("Identity of the text.")
+  id: string;
+
+  @doc("Text for analysis.")
+  text: string;
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("User input for ungroudedness detection.")
+model DetectUngroundednessInput {
+  @doc("This is optional and only used for QnA or chat task.")
+  query?: string;
+
+  @doc("Content for analysis.")
+  contents: DetectUngroundednessInputContent[];
+
+  @doc("Source information that serves as grounding source.")
+  groundingSource: string[];
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("Connection information for GPT resource.")
+model DetectUngroundednessGptxOptions {
+  @doc("Endpoint of GPT resource.")
+  azureOpenAIEndpoint: string;
+
+  @doc("Deployment id of GPT resource.")
+  deploymentId: string;
+}
+
+@added(Versions.v2023_11_30_Preview)
+@doc("The request of ungroudedness detection.")
+model DetectUngroundednessOptions {
+  @doc("Domain that the current request belongs to.")
+  domain: string;
+
+  @doc("Type of work that the customer want the API to execute.")
+  task: string;
+
+  @doc("The input to be examined for detection.")
+  input: DetectUngroundednessInput;
```

Comment:
"input" is a general concept and may cause confusion here. My suggestion is not to have too much nested object definition for customer facing APIs. You can unwarp DetectUngroundednessInput and put the three fields inside to the first layer.